syntax = "proto3";
package wedgeblock;
/**
Compile instruction:
python -m grpc_tools.protoc -I./ --python_out=. --grpc_python_out=. wedgeblock.proto
**/

service EdgeNode {
    rpc Execute (Transaction) returns (Hash1);
    rpc GetPhase2Hash (LogHash) returns (Hash2);
}

message Transaction {
    RWSet rw = 1;
}

message RWSet {
    TxnType type = 1;
    string key = 2;
    string val = 3;
}

enum TxnType {
    RO = 0;
    RW = 1;
}

message Hash1 {
    int32 logIndex = 1;
    RWSet rw = 2;
    /**
    Include Merkle tree path.
    **/
}

message LogHash {
    int32 logIndex =1;
    bytes merkleRoot = 2;
}

message Hash2 {
    bytes TxnHash = 1;
}

message LogEntry {
    int32 index = 1;

}